core-java/src/main/antlr/org/xtuml/masl/antlr/Walker.g

```diff
@@ -1355,19 +1355,18 @@ returns [PragmaDefinition def]
 
 pragmaValue
 returns [ String value ]
-                              : identifier                  { $value = $identifier.name; }
-                              | literalExpression           {
-                                                              if ( $literalExpression.exp instanceof StringLiteral ) 
+                              : expression             {
```

Comment:
Rather than using the text of the expression in the fall-through case, `evaluate()` the expression to a literal expression at the top, and then check conditions as before on this literal expression result. Conditions for the other literal types (eg IntegerLiteral) will need to be handled explicitly similarly to CharacterLiteral to do the conversion to string.